towerlib/entities/job.py

```diff
@@ -1736,3 +1736,60 @@ def module_name(self):
 #              u'notifications': u'/api/v2/ad_hoc_commands/4979/notifications/',
 #              u'relaunch': u'/api/v2/ad_hoc_commands/4979/relaunch/',
 #              u'stdout': u'/api/v2/ad_hoc_commands/4979/stdout/'},
+
+
+class Workflow(Entity):  # pylint: disable=too-many-public-methods
+    """Models the Job Template entity of ansible tower."""
+
+    def __init__(self, tower_instance, data):
+        Entity.__init__(self, tower_instance, data)
+        self._object_roles = None
+
+# add workflow_jobs
+# add workflow_job_template_nodes
+
+# step 1. Create workflow
+# step 2. Create Nodes
+# step 3. Associate nodes with each other
+
+
+    def add_workflow_nodes_to_workflow_template(self,
+                                                workflow_template,
+                                                unified_job_template):
+        """Adds workflow nodes to a workflow.
+        """
+        payload = {'unified_job_template': unified_job_template
+        }
+        url = '{api}/workflow_job_templates/{workflow_template}/workflow_nodes/'.format(api=self.api, workflow_template=workflow_template)
+        response = self.session.post(url, json=payload)
+        if not response.ok:
+            self._logger.error('Error adding workflow node to job template, response was: "%s"', response.text)
+        return Workflow(self, response.json()) if response.ok else None
+
+
+    def add_workflow_job_template_nodes(self,
+                                        workflow_template,
+                                        unified_job_template):
+        """Creates a workflow job template node.
+        """
+        payload = {'workflow_job_template': workflow_template,
+                   'unified_job_template': unified_job_template
+        }
+        url = '{api}/workflow_job_template_nodes/'.format(api=self.api)
+        response = self.session.post(url, json=payload)
+        if not response.ok:
+            self._logger.error('Error creating a workflow job template node, response was: "%s"', response.text)
+        return Workflow(self, response.json()) if response.ok else None
+
+    def add_workflow_job_template_nodes_to_workflow_job_template_node(self,
```

Comment:
Could this maybe be a bit more friendly named like "add_nodes_to_node" ? We are in a workflow context so that does not have to mentioned.